wemake_python_styleguide/visitors/ast/naming.py

```diff
@@ -172,10 +172,24 @@ class WrongVariableAssignmentVisitor(BaseNodeVisitor):
     """Finds wrong variables assignments."""
 
     def _check_assignment(self, node: ast.Assign) -> None:
-        node_value_id = getattr(node.value, 'id', None)
-        for target_node in node.targets:
-            target_node_id = getattr(target_node, 'id', None)
-            if target_node_id == node_value_id:
+        target_names = [target.id for target in node.targets
+                        if isinstance(target, ast.Name)]
+        if target_names:
+            if getattr(node.value, 'id', None) in target_names or (
```

Comment:
Each step in this `if` should be a separate variable:
```python
taget_name = getattr(node.value, 'id', None)
PLEASE_FIND_CORRECT_NAME = len(target_names) != len(set(target_names))
if target_name in target_names or PLEASE_FIND_CORRECT_NAME: ...
```

Replace `PLEASE_FIND_CORRECT_NAME` with anything you find appropriate.